### DISCUSSION TITLE
Layout swithcher and layout viewer is not working under wayland

### DISCUSSION THREAD

**Discussion by abuhuraira0099** [OpenBangla/OpenBangla-Keyboard#205] [General]
I cannot change the layout and cannot launch the layout viewer when I logged in under wayland session. But It works when using xorg.
I am using Gnome 38 on Fedora 33 Workstation. Is there any solution for this issue?

**Comment by mominul** [comment]
Sorry, I can't reproduce your issue on my Gnome Wayland session. Can you please run this command in the terminal and post the output here?
```bash
$ /usr/share/openbangla-keyboard/openbangla-gui
```
Sorry for being so late! :confounded:

**Reply by abuhuraira0099** [reply]
OpenBangla keyboard works fine. But I cannot change the layout. 
Here is the output : 
```
QSocketNotifier: Can only be used with threads started with QThread
[OpenBangla-Error]: [AutoCorrect]: Error: Couldn't open user specific AutoCorrect file!
QIODevice::read (QFile, "/home/mhmr/.local/share/openbangla-keyboard/autocorrect.json"): device not open
"/proc/8674/root"
"/proc/8674/root"

```

**Reply by mominul** [reply]
So can you see the topbar after initiating the command I mentioned above?

**Reply by abuhuraira0099** [reply]
> So can you see the topbar after initiating the command I mentioned above?

The topbar is properly launched, but I cannot move it. Every features working fine including layout viewer now. But the layout switcher still doesnt work.

**Reply by abuhuraira0099** [reply]
By the way, I tried it on Arch. It works fine on Arch under Wayland session sometimes. I think any dependency is missing in my Fedora installation, but I don't know what to do.

**Reply by mominul** [reply]
Can you view the layout switching menu by clicking the layout switching button in the topbar and select items from the menu?

**Reply by abuhuraira0099** [reply]
> Can you view the layout switching menu by clicking the layout switching button in the topbar and select items from the menu?

No. No pop up menu shown when clicking the layout switching button.

**Reply by mominul** [reply]
Can you move the topbar by dragging the OpenBangla icon button?

**Reply by abuhuraira0099** [reply]
> Can you move the topbar by dragging the OpenBangla icon button?

No. I can't.
